src/components/profile/profile-logged-in/adopted-trees.tsx

```diff
@@ -4,66 +4,104 @@ import { TertiaryButton } from "../../buttons/tertiary";
 import { TreeCard } from "./tree-card";
 import { useAuthStore } from "../../../auth/auth-store";
 import { AdoptedTreeIcon } from "../../icons/adopted-tree-icon";
+import { Skeleton } from "../../skeleton/skeleton";
+
+export interface AdoptedTreesWrapperProps {
+	sectionTitle?: string;
+}
+
+export const AdoptedTreesWrapper: React.FC<
+	AdoptedTreesWrapperProps & React.HTMLAttributes<HTMLDivElement>
+> = ({ sectionTitle, ...props }) => {
+	return (
+		<div className="md:shadow-gdk-soft mb-3 md:rounded-2xl md:border-2 md:p-7">
+			<h2 className="text-2xl font-semibold">{sectionTitle}</h2>
+			{props.children}
+		</div>
+	);
+};
 
 export const AdoptedTrees: React.FC = () => {
 	const i18n = useI18nStore().i18n();
 	const [showAllTrees, setshowAllTrees] = useState(false);
-
 	const { adoptedTreesInfo } = useAuthStore();
-	const maxTrees = adoptedTreesInfo?.length > 3 ? 4 : adoptedTreesInfo?.length;
 
-	return (
-		<div className="md:shadow-gdk-soft mb-3 md:rounded-2xl md:border-2 md:p-7">
-			<h2 className="text-2xl font-semibold">
-				{i18n.navbar.profile.overview.adoptedTrees}
-			</h2>
-			{adoptedTreesInfo.length > 0 ? (
-				<>
+	const maxTrees =
+		adoptedTreesInfo?.length ?? 0 > 3 ? 4 : adoptedTreesInfo?.length ?? 0;
```

Comment:
i think this a bit hard to read. could we simplify it? Maybe let's even extract it in a function